### PR TITLE
Added an environment variable for disabling the go profiler setup

### PR DESCRIPTION
**Pull request by bennycortese**
Mentioned in #4235 

### What's being changed:

Optional environment variable ENABLE_GO_PROFILING added for disabling the go profiler in the rest client handler

### Review checklist

- [ ] Documentation has been updated, if necessary. Link to changed documentation:
- [ ] Chaos pipeline run or not necessary. Link to pipeline:
- [ ] All new code is covered by tests where it is reasonable.
- [ ] Performance tests have been run or not necessary.

### CODE REVIEW

**Comment by bennycortese** [comment]
I agree with the CLA

**Comment by dirkkul** [comment]
Hi, we would prefer if profiling is on by default. Could you change your PR to have a DISABLE_GO_PROFILING env var?

**Comment by bennycortese** [comment]
Updated for review! @dirkkul

**Comment by dirkkul** [comment]
Thank you! There are merge conflicts, could you solve them?

**Comment by bennycortese** [comment]
@dirkkul So sorry for the delay! Things got hectic, everything should now be updated to account for merge conflicts/the latest version of main. Thank you!

File: usecases/config/environment.go [hunk 1]
```diff
@@ -169,6 +169,8 @@ func FromEnv(config *Config) error {
 		}
 	}
 
+	config.Profiling.Disabled = configbase.Enabled("DISABLE_GO_PROFILING")
```
Comment (dirkkul) [inline usecases/config/environment.go:172]: it needs to be `configbase.Enabled(os.Getenv("DISABLE_GO_PROFILING"))` - could you please double check that it works?

Comment (bennycortese) [reply usecases/config/environment.go]: Yeah so so sorry about that - I can't believe I had missed that

**Review by dirkkul — APPROVED** [review]

**Pull request merged**